consoles/serial_screen.pm

```diff
@@ -152,6 +151,45 @@ sub normalise_pattern {
     return $pattern;
 }
 
+=head2 do_read
+
+  my $num_read = do_read($buffer [, max_size => 2048][,timeout => undef]);
+
+Attempts to read up to max_size bytes from the C<<$self->{socket_fd}>> into buffer.
+The method returns as soon as some data is available, even if the given size has not been reached.
+Returns number of bytes read or undef on timeout. Note that 0 is a valid return code.
+If an failure occur the method croak.
+
+A undefined timeout will wait infinitive. A timeout of 0 just try to read once.
+=cut
+sub do_read
+{
+    my ($self, undef, %args) = @_;
+    my $buffer = '';
+    $args{timeout}  //= undef;    # wait till data is available
```

Comment:
looks strange , it will be undef anyway . what the purpose of this line ?

Reply:
Yeah true, I added it to make the reader of that function aware that timeout is a valid parameter.
And to add the comment, that `undef` is treated as infinite waiting.